sound/soc/sof/sof-audio.h

```diff
@@ -387,6 +389,22 @@ struct snd_sof_widget {
 	int num_tuples;
 	struct snd_sof_tuple *tuples;
 
+	/* TODO: populate these 2 in the widget ipc_setup op after parsing the tokens from topology */
+	u32 num_source_pins;
+	u32 num_sink_pins;
+
+	struct ida source_queue_ida;
+	struct ida sink_queue_ida;
+
+	/*
+	 * array containing names of connecting source/sink widgets for size equal to the num
+	 * source/sink pins
+	 * TODO: This array should be populated when parsing the pin_widget_binding tokens from
+	 * topology
+	 */
+	char *widget_source_pin_binding[MAX_NAME_SIZE];
```

Comment:
Why not keep these are char **. The max name size for widgets is not defined anywhere in soc-topology and could be larger than 64 bytes.